galaxy/pulp/tasks.py

```diff
@@ -164,3 +166,23 @@ def _log_collection_loaded(coll_info):
             content.name,
             getattr(content, 'scores'),
         ))
+
+
+def _write_collection_tags(collection, version, i_collection):
+    '''Update tags at collection-level, only if highest version'''
+
+    log.debug('highest_version={}'.format(collection.highest_version.version))
+    if collection.highest_version != version:
+        return
+
+    for tag in i_collection.collection_info.tags:
```

Comment:
Better use bulk operations here.

Reply:
@cutwater I put a code comment to add bulk operations, should occur for role tags also as the same code is used in both places

Reply:
Can you please make it part of this PR.

Reply:
The bulk operations have been added to this PR